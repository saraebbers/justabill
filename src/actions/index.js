export const getBills = (bills) => ({
  type: 'GET_BILLS',
  bills
})

export const addFavorite = (bill_id) => ({
  type: 'ADD_FAVORITE',
  bill_id
})

export const removeFavorite = (bill_id) => ({
  type: 'REMOVE_FAVORITE',
  bill_id
})

export const hasErrored = (errorMessage) => ({
  type: 'HAS_ERRORED',
  errorMessage
})

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})