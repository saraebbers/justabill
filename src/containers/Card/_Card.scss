.card {
  background-color: #DBCFB0;
  margin: 15px 5px;
  padding: 10px;
  border: solid #718F94 4px;
  color: #420C14;
  overflow: scroll;
  height: 350px;

  &.active {
    background-color: #718F94;
  }


  &:hover {
    background-color: #ADA380;
    color: #151E3F;
  }

  &::-webkit-scrollbar, 
  &::-webkit-scrollbar-corner {
      background-color: transparent;
    }

  h4 {
    text-align: center;
    font-size: 1.25rem;
    color: #151E3F;
    padding: 5px;
  }

  p {
    font-size: 1rem;
  }

  h5 {
    font-size: 1rem;
    text-align: center;
  }

}