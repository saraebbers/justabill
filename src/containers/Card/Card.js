import React, { Component } from 'react';
import '../../index.scss';
import { addFavorite, removeFavorite } from '../../actions';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';


export class Card extends Component {

  handleAddFavorite = () => {
    const { id, addFavorite } = this.props
    addFavorite(id)

  }

  handleRemoveFavorite = () => {
    const { id, removeFavorite } = this.props
    removeFavorite(id)
  }


  render() { 
    let { id, title, sponsor, major_action, major_action_date, summary, bill_uri, favorites } = this.props
    let button
    
    if(summary === '') {
      summary = 'A summary of this bill is not currently available'
    }

    if(favorites.includes(id)) {
      button= 
        <button 
          onClick={() => this.handleRemoveFavorite()}
          className='fav'
          >Remove bill {id} from Favorites</button>
    } else {
       button= 
        <button 
          onClick={() => this.handleAddFavorite()}
          className='not-fav'
          >Save bill {id} as Favorite</button>
    }

    return(
      <div className='card'>
        {button}
        <h4>{title}</h4>
        <h5>Sponsored by: {sponsor} </h5>
        <h5>{major_action}</h5>
        <h5>Date of last Major Action: {major_action_date}</h5>
        <h5>Summary: </h5>
        <p>{summary}</p>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  favorites: state.favorites
})

export const mapDispatchToProps = (dispatch) => ({
  addFavorite: (id) => dispatch(addFavorite(id)),
  removeFavorite: (id) => dispatch(removeFavorite(id))
})

Card.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  sponsor: PropTypes.string.isRequired,
  major_action: PropTypes.string.isRequired,
  major_action_date: PropTypes.string.isRequired,
  summary: PropTypes.string.isRequired,
  bill_uri: PropTypes.string.isRequired,
  addFavorite: PropTypes.func.isRequired,
  removeFavorite: PropTypes.func.isRequired,
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Card));