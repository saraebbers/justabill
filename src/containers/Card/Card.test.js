import React from 'react';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps, Card } from './Card';
import { addFavorite, removeFavorite } from '../../actions/index';

describe('Card', () => {
  let mockFavorites;
  let mockId 
  let mockTitle
  let mockSponsor
  let mockMajor_action
  let mockMajor_action_date
  let mockSummary
  let mockBill_uri
  let mockAddFavorite
  let mockRemoveFavorite


  beforeEach(() => {
    mockFavorites=['hr-45123', 'hr-7653']
    mockId = 'hr-7654'
    mockTitle = 'A very long string'
    mockSponsor = 'Wonder Woman'
    mockMajor_action = 'Bill became Law'
    mockMajor_action_date = '12.6.2016'
    mockSummary = 'To begin my life with the beginning of my life I must record that I was born'
    mockBill_uri = 'https://something'
    mockAddFavorite = jest.fn()
    mockRemoveFavorite = jest.fn()
  })

  describe('Card Container', () => {
    it('should match the snapshot if summary is empty', () => {
      mockSummary = ''
      const wrapper = shallow(<Card favorites={mockFavorites} id={mockId} title={mockTitle} sponsor={mockSponsor} major_action={mockMajor_action} major_action_date={mockMajor_action_date} summary={mockSummary} bill_uri={mockBill_uri} addFavorite={mockAddFavorite} removeFavorite={mockRemoveFavorite}/>)
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot if summary is provided and id is not in the favorites array', () => {
      const wrapper = shallow(<Card favorites={mockFavorites} id={mockId} title={mockTitle} sponsor={mockSponsor} major_action={mockMajor_action} major_action_date={mockMajor_action_date} summary={mockSummary} bill_uri={mockBill_uri} addFavorite={mockAddFavorite} removeFavorite={mockRemoveFavorite}/>)
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot if the id is included in favorites array', () => {
      mockId= 'hr-45123'
      const wrapper = shallow(<Card favorites={mockFavorites} id={mockId} title={mockTitle} sponsor={mockSponsor} major_action={mockMajor_action} major_action_date={mockMajor_action_date} summary={mockSummary} bill_uri={mockBill_uri} addFavorite={mockAddFavorite} removeFavorite={mockRemoveFavorite}/>)
      expect(wrapper).toMatchSnapshot()
    })

  })

  describe('mapStateToProps', () => {
    it('should return a props object with the correct information from state parsed', () => {
      const mockBillArray = [{congress: 116, bills: [{id: 'hr123', name:'sara', somethingelse: 'xxx'}, {id: 'hr234', name:'David', somethingelse: 'xxx'}]}]
      const mockState = {
        bills: mockBillArray,
        errorMessage: 'this is an error message',
        favorites: mockFavorites,
        somethingElse: 'something else'
      }
      const expected = {
        favorites: mockFavorites
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with addFavorite when addFavorite is called', () => {
      const mockDispatch = jest.fn()      
      const actionToDispatch = addFavorite('hr-8989')
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addFavorite('hr-8989')
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with removeFavorite when removeFavorite is called', () => {
      const mockDispatch = jest.fn()      
      const actionToDispatch = removeFavorite('hr-7653')
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.removeFavorite('hr-7653')
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

  })

  describe('handleAddFavorite', () => {
    it('should call addFavorite with the correct params when a button is clicked and the id is not in the favorites array', () => {
      mockId = 'hr-9'
      const wrapper = shallow(<Card favorites={mockFavorites} id={mockId} title={mockTitle} sponsor={mockSponsor} major_action={mockMajor_action} major_action_date={mockMajor_action_date} summary={mockSummary} bill_uri={mockBill_uri} addFavorite={mockAddFavorite} removeFavorite={mockRemoveFavorite}/>)
      wrapper.find('.not-fav').simulate('click')
      expect(mockAddFavorite).toHaveBeenCalled()
    })
  })

  describe('handleRemoveFavorite', () => {
    it('should call removeFavorite with the correct params when a button is clicked and the id is in the favorites array', () => {
      mockId = 'hr-7653'
      const wrapper = shallow(<Card favorites={mockFavorites} id={mockId} title={mockTitle} sponsor={mockSponsor} major_action={mockMajor_action} major_action_date={mockMajor_action_date} summary={mockSummary} bill_uri={mockBill_uri} addFavorite={mockAddFavorite} removeFavorite={mockRemoveFavorite}/>)
      wrapper.find('.fav').simulate('click')
      expect(mockRemoveFavorite).toHaveBeenCalled()
    })
  })


})
