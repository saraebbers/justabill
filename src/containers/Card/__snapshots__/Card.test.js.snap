// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card Container should match the snapshot if summary is empty 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorite={[MockFunction]}
    bill_uri="https://something"
    favorites={
      Array [
        "hr-45123",
        "hr-7653",
      ]
    }
    id="hr-7654"
    major_action="Bill became Law"
    major_action_date="12.6.2016"
    removeFavorite={[MockFunction]}
    sponsor="Wonder Woman"
    summary=""
    title="A very long string"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="not-fav"
          onClick={[Function]}
        >
          Save bill 
          hr-7654
           as Favorite
        </button>,
        <h4>
          A very long string
        </h4>,
        <h5>
          Sponsored by: 
          Wonder Woman
           
        </h5>,
        <h5>
          Bill became Law
        </h5>,
        <h5>
          Date of last Major Action: 
          12.6.2016
        </h5>,
        <h5>
          Summary: 
        </h5>,
        <p>
          A summary of this bill is not currently available
        </p>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Save bill ",
            "hr-7654",
            " as Favorite",
          ],
          "className": "not-fav",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Save bill ",
          "hr-7654",
          " as Favorite",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A very long string",
        },
        "ref": null,
        "rendered": "A very long string",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Sponsored by: ",
            "Wonder Woman",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Sponsored by: ",
          "Wonder Woman",
          " ",
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bill became Law",
        },
        "ref": null,
        "rendered": "Bill became Law",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Date of last Major Action: ",
            "12.6.2016",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Date of last Major Action: ",
          "12.6.2016",
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Summary: ",
        },
        "ref": null,
        "rendered": "Summary: ",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A summary of this bill is not currently available",
        },
        "ref": null,
        "rendered": "A summary of this bill is not currently available",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="not-fav"
            onClick={[Function]}
          >
            Save bill 
            hr-7654
             as Favorite
          </button>,
          <h4>
            A very long string
          </h4>,
          <h5>
            Sponsored by: 
            Wonder Woman
             
          </h5>,
          <h5>
            Bill became Law
          </h5>,
          <h5>
            Date of last Major Action: 
            12.6.2016
          </h5>,
          <h5>
            Summary: 
          </h5>,
          <p>
            A summary of this bill is not currently available
          </p>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Save bill ",
              "hr-7654",
              " as Favorite",
            ],
            "className": "not-fav",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Save bill ",
            "hr-7654",
            " as Favorite",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A very long string",
          },
          "ref": null,
          "rendered": "A very long string",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Sponsored by: ",
              "Wonder Woman",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Sponsored by: ",
            "Wonder Woman",
            " ",
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bill became Law",
          },
          "ref": null,
          "rendered": "Bill became Law",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Date of last Major Action: ",
              "12.6.2016",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Date of last Major Action: ",
            "12.6.2016",
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Summary: ",
          },
          "ref": null,
          "rendered": "Summary: ",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A summary of this bill is not currently available",
          },
          "ref": null,
          "rendered": "A summary of this bill is not currently available",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card Card Container should match the snapshot if summary is provided and id is not in the favorites array 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorite={[MockFunction]}
    bill_uri="https://something"
    favorites={
      Array [
        "hr-45123",
        "hr-7653",
      ]
    }
    id="hr-7654"
    major_action="Bill became Law"
    major_action_date="12.6.2016"
    removeFavorite={[MockFunction]}
    sponsor="Wonder Woman"
    summary="To begin my life with the beginning of my life I must record that I was born"
    title="A very long string"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="not-fav"
          onClick={[Function]}
        >
          Save bill 
          hr-7654
           as Favorite
        </button>,
        <h4>
          A very long string
        </h4>,
        <h5>
          Sponsored by: 
          Wonder Woman
           
        </h5>,
        <h5>
          Bill became Law
        </h5>,
        <h5>
          Date of last Major Action: 
          12.6.2016
        </h5>,
        <h5>
          Summary: 
        </h5>,
        <p>
          To begin my life with the beginning of my life I must record that I was born
        </p>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Save bill ",
            "hr-7654",
            " as Favorite",
          ],
          "className": "not-fav",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Save bill ",
          "hr-7654",
          " as Favorite",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A very long string",
        },
        "ref": null,
        "rendered": "A very long string",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Sponsored by: ",
            "Wonder Woman",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Sponsored by: ",
          "Wonder Woman",
          " ",
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bill became Law",
        },
        "ref": null,
        "rendered": "Bill became Law",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Date of last Major Action: ",
            "12.6.2016",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Date of last Major Action: ",
          "12.6.2016",
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Summary: ",
        },
        "ref": null,
        "rendered": "Summary: ",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "To begin my life with the beginning of my life I must record that I was born",
        },
        "ref": null,
        "rendered": "To begin my life with the beginning of my life I must record that I was born",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="not-fav"
            onClick={[Function]}
          >
            Save bill 
            hr-7654
             as Favorite
          </button>,
          <h4>
            A very long string
          </h4>,
          <h5>
            Sponsored by: 
            Wonder Woman
             
          </h5>,
          <h5>
            Bill became Law
          </h5>,
          <h5>
            Date of last Major Action: 
            12.6.2016
          </h5>,
          <h5>
            Summary: 
          </h5>,
          <p>
            To begin my life with the beginning of my life I must record that I was born
          </p>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Save bill ",
              "hr-7654",
              " as Favorite",
            ],
            "className": "not-fav",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Save bill ",
            "hr-7654",
            " as Favorite",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A very long string",
          },
          "ref": null,
          "rendered": "A very long string",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Sponsored by: ",
              "Wonder Woman",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Sponsored by: ",
            "Wonder Woman",
            " ",
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bill became Law",
          },
          "ref": null,
          "rendered": "Bill became Law",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Date of last Major Action: ",
              "12.6.2016",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Date of last Major Action: ",
            "12.6.2016",
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Summary: ",
          },
          "ref": null,
          "rendered": "Summary: ",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "To begin my life with the beginning of my life I must record that I was born",
          },
          "ref": null,
          "rendered": "To begin my life with the beginning of my life I must record that I was born",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card Card Container should match the snapshot if the id is included in favorites array 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorite={[MockFunction]}
    bill_uri="https://something"
    favorites={
      Array [
        "hr-45123",
        "hr-7653",
      ]
    }
    id="hr-45123"
    major_action="Bill became Law"
    major_action_date="12.6.2016"
    removeFavorite={[MockFunction]}
    sponsor="Wonder Woman"
    summary="To begin my life with the beginning of my life I must record that I was born"
    title="A very long string"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="fav"
          onClick={[Function]}
        >
          Remove bill 
          hr-45123
           from Favorites
        </button>,
        <h4>
          A very long string
        </h4>,
        <h5>
          Sponsored by: 
          Wonder Woman
           
        </h5>,
        <h5>
          Bill became Law
        </h5>,
        <h5>
          Date of last Major Action: 
          12.6.2016
        </h5>,
        <h5>
          Summary: 
        </h5>,
        <p>
          To begin my life with the beginning of my life I must record that I was born
        </p>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Remove bill ",
            "hr-45123",
            " from Favorites",
          ],
          "className": "fav",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Remove bill ",
          "hr-45123",
          " from Favorites",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A very long string",
        },
        "ref": null,
        "rendered": "A very long string",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Sponsored by: ",
            "Wonder Woman",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Sponsored by: ",
          "Wonder Woman",
          " ",
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bill became Law",
        },
        "ref": null,
        "rendered": "Bill became Law",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Date of last Major Action: ",
            "12.6.2016",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Date of last Major Action: ",
          "12.6.2016",
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Summary: ",
        },
        "ref": null,
        "rendered": "Summary: ",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "To begin my life with the beginning of my life I must record that I was born",
        },
        "ref": null,
        "rendered": "To begin my life with the beginning of my life I must record that I was born",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="fav"
            onClick={[Function]}
          >
            Remove bill 
            hr-45123
             from Favorites
          </button>,
          <h4>
            A very long string
          </h4>,
          <h5>
            Sponsored by: 
            Wonder Woman
             
          </h5>,
          <h5>
            Bill became Law
          </h5>,
          <h5>
            Date of last Major Action: 
            12.6.2016
          </h5>,
          <h5>
            Summary: 
          </h5>,
          <p>
            To begin my life with the beginning of my life I must record that I was born
          </p>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Remove bill ",
              "hr-45123",
              " from Favorites",
            ],
            "className": "fav",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Remove bill ",
            "hr-45123",
            " from Favorites",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A very long string",
          },
          "ref": null,
          "rendered": "A very long string",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Sponsored by: ",
              "Wonder Woman",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Sponsored by: ",
            "Wonder Woman",
            " ",
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bill became Law",
          },
          "ref": null,
          "rendered": "Bill became Law",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Date of last Major Action: ",
              "12.6.2016",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Date of last Major Action: ",
            "12.6.2016",
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Summary: ",
          },
          "ref": null,
          "rendered": "Summary: ",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "To begin my life with the beginning of my life I must record that I was born",
          },
          "ref": null,
          "rendered": "To begin my life with the beginning of my life I must record that I was born",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
